src/elements/content-sidebar/withSidebarAnnotations.js

```diff
@@ -0,0 +1,179 @@
+            getAnnotationsPath: noop,
+            onVersionChange: noop,
+        };
+
+        static displayName: ?string;
+
+        props: Props;
+
+        sidebarPanels = React.createRef();
+
+        constructor(props) {
+            super(props);
+
+            this.redirectDeeplinkedAnnotation();
+        }
+
+        redirectDeeplinkedAnnotation = () => {
+            const { file, getAnnotationsPath, getAnnotationsMatchPath, history, location } = this.props;
+            const match = getAnnotationsMatchPath(location);
+            const annotationId = getProp(match, 'params.annotationId');
+            const currentFileVersionId = getProp(file, 'file_version.id');
+            const fileVersionId = getProp(match, 'params.fileVersionId');
+
+            if (fileVersionId && fileVersionId !== currentFileVersionId) {
+                history.replace(getAnnotationsPath(currentFileVersionId, annotationId));
+            }
+        };
+
+        componentDidUpdate(prevProps: Props) {
+            const { annotatorState, getAnnotationsMatchPath, location }: Props = this.props;
+            const { annotatorState: prevAnnotatorState, location: prevLocation }: Props = prevProps;
+            const { activeAnnotationId, annotation } = annotatorState;
+            const { activeAnnotationId: prevActiveAnnotationId, annotation: prevAnnotation } = prevAnnotatorState;
+
+            const match = getAnnotationsMatchPath(location);
+            const prevMatch = getAnnotationsMatchPath(prevLocation);
+            const fileVersionId = getProp(match, 'params.fileVersionId');
+            const hasActiveAnnotationChanged = prevActiveAnnotationId !== activeAnnotationId;
+            const isAnnotationsPath = !!match;
+            const isTransitioningToAnnotationPath = activeAnnotationId && !isAnnotationsPath;
+            const prevFileVersionId = getProp(prevMatch, 'params.fileVersionId');
+
+            if (annotation && prevAnnotation !== annotation) {
+                this.addAnnotation();
+            }
+
+            // Active annotation id changed. If location is currently an annotation path or
+            // if location is not currently an annotation path but the active annotation id
+            // transitioned from falsy to truthy, update the location accordingly
+            if (hasActiveAnnotationChanged && (isAnnotationsPath || isTransitioningToAnnotationPath)) {
+                this.updateActiveAnnotation();
+            }
+
+            if (prevFileVersionId !== fileVersionId) {
+                this.updateActiveVersion();
+            }
+        }
+
+        addAnnotation() {
+            const {
+                annotatorState: { action, annotation, meta },
+                api,
+                currentUser,
+                file,
+                fileId,
+                getAnnotationsMatchPath,
+                isOpen,
+                location,
+            } = this.props;
+            const { requestId } = meta || {};
```

Comment:
If `meta` and `requestId` are undefined, should we abort? We use the latter as the object id below.